content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-contract.md

```diff
@@ -206,94 +216,35 @@ class MarsVoucherContract : Contract {
             is BoardingTicketContract.Commands.RedeemTicket-> requireThat {
                 //Transaction verification will happen in BoardingTicket Contract
             }
+            is Commands.Transfer -> requireThat {
```

Comment:
shouldn't we also check that its the current holder initiating the transfer

Reply:
@peterli-r3 What do you think?

Reply:
Sure. Will add it to the code.

Reply:
Initiator check added in the GiftVoucherToFriend Flow.

Reply:
Okay I'll address this in a separate PR. Thanks.